pydicom/dataset.py

```diff
@@ -1541,19 +1541,22 @@ def _convert_pixel_data_without_handler(self) -> None:
                 )
                 last_exception = exc
 
-        # The only way to get to this point is if we failed to get the pixel
-        #   array because all suitable handlers raised exceptions
-        self._pixel_array = None
-        self._pixel_id = {}
-
-        logger.info(
-            "Unable to decode the pixel data using the following handlers: {}."
-            "Please see the list of supported Transfer Syntaxes in the "
-            "pydicom documentation for alternative packages that might "
-            "be able to decode the data"
-            .format(", ".join([str(hh) for hh in available_handlers]))
-        )
-        raise last_exception  # type: ignore[misc]
+        if last_exception is None:
+            pass  # FIXME: raise some exception?
```

Comment:
This can never happen, as mentioned in the comment in the else branch. If the check is needed for some static checker, an `assert last_exception is not None` could help.

Reply:
It can happen if `available_handlers` is empty.

Reply:
This is already handled above - it will raise `RuntimeError`.